db/models/user.js

```diff
@@ -1,28 +1,41 @@
 'use strict'
+const Address = require('./address.js')
 
 // bcrypt docs: https://www.npmjs.com/package/bcrypt
 const bcrypt = require('bcryptjs')
-    , {STRING, VIRTUAL} = require('sequelize')
+    , {BOOLEAN, STRING, VIRTUAL} = require('sequelize')
 
-module.exports = db => db.define('users', {
-  name: STRING,
+const userSchema = {
+  userName: STRING,
   email: {
     type: STRING,
+    unique: true,
     validate: {
       isEmail: true,
-      notEmpty: true,
     }
   },
-
+  isAdmin: {
+    type: BOOLEAN,
+    defaultValue: false
+  },
   // We support oauth, so users may or may not have passwords.
   password_digest: STRING, // This column stores the hashed password in the DB, via the beforeCreate/beforeUpdate hooks
   password: VIRTUAL // Note that this is a virtual, and not actually stored in DB
-}, {
+}
+
+const optionMethods = {
   indexes: [{fields: ['email'], unique: true}],
   hooks: {
     beforeCreate: setEmailAndPassword,
     beforeUpdate: setEmailAndPassword,
   },
+
+  getterMethods: {
+    // checks if user is a guest or not
```

Comment:
isGuest getter method may need more checks...

guest needs to be a more distinguished category separate from user